auxx/src/Lang/Interpreter.hs

```diff
@@ -35,10 +37,16 @@ eval = \case
 -- | Evaluage a group of expressions and take the result
 -- of the last one. Equivalent to @fmap NE.last . traverse eval@,
 -- but is single-pass.
-evalExprGroup :: NonEmpty (Expr (CommandProc m)) -> EvalT m Value
-evalExprGroup (x :| xs) = case nonEmpty xs of
-    Nothing  -> eval x
-    Just xs' -> eval x *> evalExprGroup xs'
+evalExprGroup :: AtLeastTwo (Expr (CommandProc m)) -> EvalT m Value
+evalExprGroup altExprs = case (Lang.toList_ altExprs) of
```

Comment:
Instead of calling `toList_` and then doing an unsafe pattern match, just don't call it and do a safe pattern match.